tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/AsyncToSyncMethodTransformerTests.cs

```diff
@@ -220,6 +220,78 @@ internal sealed class AsyncToSyncMethodTransformerTests {
 		Assert.AreEqual( "[Blocking] void Bar() { TryBaz( input,out int result ); }", actual.Value.ToFullString() );
 	}
 
+	[Test]
+	public void TaskToVoidReturnTypeInvocation() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return BarAsync(2); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { { Bar(2); return; } }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeNull() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return null; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { return; }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeDoubleInvocation() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { if( QuuxAsync() ) { return BarAsync(2); } return BarAsync(4); }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { if( Quux() ) { { Bar(2); return; } } { Bar(4); return; } }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeIncrement() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return bar++; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { { bar++; return; } }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypePreDecrement() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return --bar; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { { --bar; return; } }", actual.Value.ToFullString() );
+	}
+
+	[Test]
+	public void TaskToVoidReturnTypeMemberAccess() {
+		var actual = Transform( @"[GenerateSync] Task FooAsync() { return Baz.bar; }" );
+
+		Assert.IsTrue( actual.Success );
+		Assert.IsEmpty( actual.Diagnostics );
+		Assert.AreEqual( "[Blocking] void Foo() { return; }", actual.Value.ToFullString() );

```

Comment:
👍 I mentioned in DM that technically property getters could do evil here, and we'd be changing the semantics in the generated code silently... but I think that's OK, it would be insane to do that.